jsontext/pointer.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build goexperiment.rangefunc
+
+package jsontext
+
+import "iter"
+
+// Tokens returns an iterator for each reference token in the JSON pointer,
```

Comment:
I find this phrasing of this first clause strangely confusing. When I read it I think this will return multiple iterators "an iterator for each reference token". Based on the language used in https://github.com/golang/go/issues/61899 to document things that return iterators perhaps:

> Tokens returns an iterator over the reference tokens in the JSON pointer,